servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H1ProtocolConfigBuilder.java

```diff
@@ -130,14 +131,39 @@ public H1ProtocolConfigBuilder trailersEncodedSizeEstimate(final int trailersEnc
         return this;
     }
 
+    /**
+     * Defines if an HTTP/1.1 response with <a href="https://tools.ietf.org/html/rfc7230#section-6.1">
+     * Connection: close</a> and <a href="https://tools.ietf.org/html/rfc7230#section-3.3.1">
+     * Transfer-Encoding: chunked</a> headers that does not start reading the
+     * <a href="https://tools.ietf.org/html/rfc7230#section-4.1">chunked-body</a> before server closes the connection
+     * should be considered as a legit response.
+     * <p>
+     * While this use-case is not supported by <a href="https://tools.ietf.org/html/rfc7230#section-3.3.3">RFC 7230</a>,
+     * some older server implementations may use connection closure as an indicator of message completion even if
+     * {@code Transfer-Encoding: chunked} header is present:
+     * <pre>{@code
+     *     HTTP/1.1 200 OK
+     *     Content-Type: text/plain
+     *     Transfer-Encoding: chunked
```

Comment:
I am thinking if we need to make a special mention of `Transfer-Encoding: chunked` here? 

Can we have this option generally as `allowPrematureClosureWithNoPayload()`?

We can then generally say that if we see connection closure and no payload was received we assume there was no payload sent, unless `contentLength > 0`.

Reply:
> If a response has neither Content-Length nor Transfer-Encoding: chunked this is a legit use-case

Yes I do understand that it is a valid use-case and we support it today. My suggestion is to be less specific when describing the nature of this flag (either in name or javadoc).

Reply:
As in the comment above, I will prefer to see exceptions to be specific.
Since other use-cases are legit and do not require this exception, it will confuse users and some users will enable this exception without need for that.

Reply:
Discussed offline and decided to avoid mentioning chunked encoding in the API and javadoc to simplify understanding of the exception.
Done in https://github.com/apple/servicetalk/pull/1060/commits/64fb4cc1ab4a8c3316a094560bd531dd236615f3